matter_server/server/vendor_info.py

```diff
@@ -0,0 +1,97 @@
+from ..common.models import APICommand, VendorInfo as VendorInfoModel
+
+if TYPE_CHECKING:
+    from .server import MatterServer
+
+LOGGER = logging.getLogger(__name__)
+PRODUCTION_URL = "https://on.dcl.csa-iot.org"
+DATA_KEY_VENDOR_INFO = "vendor_info"
+
+
+class VendorInfo:
+    """Fetches vendor info from the CSA and handles api calls to get it."""
+
+    def __init__(self, server: MatterServer):
+        """Initialize the vendor info."""
+        self._data: dict[int, VendorInfoModel] = {}
+        self._server = server
+
+    async def start(self) -> None:
+        """Async initialize the vendor info."""
+        self._load_vendors()
+        await self._fetch_vendors()
+        self._save_vendors()
+
+    def _load_vendors(self) -> None:
+        """Load vendor info from storage."""
+        LOGGER.info("Loading vendor info from storage.")
+        vendor_count = 0
+        data = self._server.storage.get(DATA_KEY_VENDOR_INFO, {})
+        for vendor_id, vendor_info in data.items():
+            self._data[vendor_id] = dataclass_from_dict(VendorInfoModel, vendor_info)
+            vendor_count += 1
+        LOGGER.info("Loaded %s vendors from storage.", vendor_count)
+
+    async def _fetch_vendors(self) -> None:
+        """Fetch the vendor names from the CSA."""
+        LOGGER.info("Fetching the latest vendor info from DCL.")
+        vendors: dict[int, VendorInfoModel] = {}
+        try:
+            async with ClientSession(raise_for_status=True) as session:
+                async with session.get(
+                    f"{PRODUCTION_URL}/dcl/vendorinfo/vendors"
+                ) as response:
+                    data = await response.json()
+                    for vendorinfo in data["vendorInfo"]:
+                        vendors[vendorinfo["vendorID"]] = VendorInfoModel(
+                            vendor_id=vendorinfo["vendorID"],
+                            vendor_name=vendorinfo["vendorName"],
+                            company_legal_name=vendorinfo["companyLegalName"],
+                            company_preferred_name=vendorinfo["companyPreferredName"],
+                            vendor_landing_page_url=vendorinfo["vendorLandingPageURL"],
+                            creator=vendorinfo["creator"],
+                        )
+        except ClientError as err:
+            LOGGER.error("Unable to fetch vendor info from DCL: %s", err)
+        else:
+            LOGGER.info("Fetched %s vendors from DCL.", len(vendors))
+
+        self._data.update(vendors)
+
+    def _save_vendors(self) -> None:
+        """Save vendor info to storage."""
+        LOGGER.info("Saving vendor info to storage.")
+        self._server.storage.set(
+            DATA_KEY_VENDOR_INFO,
+            {
+                vendor_id: dataclass_to_dict(vendor_info)
+                for vendor_id, vendor_info in self._data.items()
+            },
+        )
+
+    @api_command(APICommand.GET_VENDOR_NAMES)
```

Comment:
I think a new command requires a bump of schema version. The server needs to support this command if a client wants to send the command to the server.

Reply:
sharp as always Martin. I'll bump the schema